source/developer/fx-guidelines.rst

```diff
@@ -181,8 +192,106 @@ Display: Icons
     a. File a bug if the icon is difficult to understand or has cosmetic defects (size, blur, etc.)
     b. File a bug when an icon doesn’t obviously indicate the underlying feature
 
+---------------------------
+
+Display: Mentions(@) icon
+---------------------------
+
+The username of any user should be preceded by the @ icon at all places except for the Direct messages list in the Left hand sidebar, and for any post or comment made by the user.
+
+Examples of some places where an @ icon should be preceeded before the username are as follows:
+
+- User list in the system console.
+- Autocomplete.
+- Members list for the current channel.
+
+---------------------------
+
+Display: Number of Choices
+--------------------------
+
+To simplify decisions, when practical, limit the number of choices to 3 or 4, and add separators or headings between logical groups. See `Hick’s Law <https://en.wikipedia.org/wiki/Hick%27s_law>`_ for background on why this helps.
+
+Example:
+
+    Incorrect:
+        No clear separation between distinct options.
+
+        ..  image:: ../images/choices1.png
+            :alt: No separation
+            :width: 500 px
+
+    Correct:
+        A clear separation between distinct options.
+
+        ..  image:: ../images/choices2.png
+            :alt: Clear separation
+            :width: 500 px
+
+-----
+
+Feedback: Error Messages
+------------------------
+
+On occasion, Mattermost users may encounter a problem that prevents them or the Mattermost system from successfully completing a task. The unexpected user behaviour or system response should be communicated to the user through an error message, and should follow the design principles of forgiving and obvious.
+
+Error messages should:
+
+- State the encountered problem with the component in the title (e.g. Team URL Not Found).
+- Describe very briefly why the error happened under the title.
+- When possible, have a link to direct users to help complete their original task or return to the previous state.
+- Give recommendations/solution(s) for the next course of action.
+- Be consistent with the Mattermost voice and look professional.
+- Be avoided when possible (by avoiding the error condition).
+
+Error message should not:
+
+- Be in red or in capital letters.
+- Say or have the term “Error”.
+- Be lengthy.
+- Be generic.
+- End in an exclamation point.
+
+**Examples:**
+
+    This is an example of a good error message:
+
+    ..  image:: ../images/error1.png
+        :alt: Good Error Message
+
+    This is an example of a bad error message:
+
+    ..  image:: ../images/error2.png
+        :alt: Bad Error Message
+
 -----
 
+Input: Fields
+-------------
+
+Users should enter information into fields without much thinking. ENTER button on last
+input field should trigger default dialog button. When last input field in a series has
+focus and user hits ENTER it should trigger the default button in the dialog.
```

Comment:
Should "Enter" be "Return"? Either way it should be lower case to match with other docs